[watchOS] Text input list view controller is not legible when presented over a transparent background

whsieh opened the pull request:
#### 3218bc287cc3297da11f4d198fe6fe561f439d1e
<pre>
[watchOS] Text input list view controller is not legible when presented over a transparent background
<a href="https://bugs.webkit.org/show_bug.cgi?id=270821">https://bugs.webkit.org/show_bug.cgi?id=270821</a>
<a href="https://rdar.apple.com/118537615">rdar://118537615</a>

Reviewed by Abrar Rahman Protyasha and Tim Horton.

Explicitly set the input view controllers&apos; views&apos; background colors to be `+systemBackgroundColor`.
Currently, these views are transparent, which works well in practice (since they present over a
black background anyways), but for certain clients that use a transparent background, this can cause
system UI to show up from behind the input view controllers.

* Source/WebKit/Platform/spi/watchos/PepperUICoreSPI.h:

Drive-by fix: remove some dead code, now that `-listView` is no longer implemented by the base
class when presenting a single or multi-select menu. Instead, reloading the `-collectionView` is
sufficient.

* Source/WebKit/UIProcess/ios/forms/WKDatePickerViewController.mm:
(-[WKDatePickerViewController viewDidLoad]):
* Source/WebKit/UIProcess/ios/forms/WKNumberPadViewController.mm:
(-[WKNumberPadViewController viewDidLoad]):
* Source/WebKit/UIProcess/ios/forms/WKSelectMenuListViewController.mm:
(-[WKSelectMenuListViewController viewDidLoad]):
(-[WKSelectMenuListViewController didSelectListItemAtIndexPath:]):
(-[WKSelectMenuListViewController cellForListItem:]): Deleted.
* Source/WebKit/UIProcess/ios/forms/WKTextInputListViewController.mm:
(-[WKTextInputListViewController viewDidLoad]):
* Source/WebKit/UIProcess/ios/forms/WKTimePickerViewController.mm:
(-[WKTimePickerViewController viewDidLoad]):

Canonical link: <a href="https://commits.webkit.org/275947@main">https://commits.webkit.org/275947@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/32c390be56eef55191dd0ddcfd3c67205737deff

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/43358 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/22388 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/45768 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/45993 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/39484 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/26143 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/19806 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/35837 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/43932 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/19405 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/37340 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/16815 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/16981 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/1414 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/39538 "Passed tests") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/38717 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/47537 "Built successfully") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/18272 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/50/builds/15029 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/42622 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/19809 "Built successfully") | | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/41282 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/9644 "Built successfully and passed tests") | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/19988 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/19440 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aprotyas reviewed (approved):
LGTM, I wonder if we could just do this for any view controller inheriting from `PUICQuickboardViewController`.

whsieh committed:
[watchOS] Text input list view controller is not legible when presented over a transparent background
https://bugs.webkit.org/show_bug.cgi?id=270821
rdar://118537615

Reviewed by Abrar Rahman Protyasha and Tim Horton.

Explicitly set the input view controllers' views' background colors to be `+systemBackgroundColor`.
Currently, these views are transparent, which works well in practice (since they present over a
black background anyways), but for certain clients that use a transparent background, this can cause
system UI to show up from behind the input view controllers.

* Source/WebKit/Platform/spi/watchos/PepperUICoreSPI.h:

Drive-by fix: remove some dead code, now that `-listView` is no longer implemented by the base
class when presenting a single or multi-select menu. Instead, reloading the `-collectionView` is
sufficient.

* Source/WebKit/UIProcess/ios/forms/WKDatePickerViewController.mm:
(-[WKDatePickerViewController viewDidLoad]):
* Source/WebKit/UIProcess/ios/forms/WKNumberPadViewController.mm:
(-[WKNumberPadViewController viewDidLoad]):
* Source/WebKit/UIProcess/ios/forms/WKSelectMenuListViewController.mm:
(-[WKSelectMenuListViewController viewDidLoad]):
(-[WKSelectMenuListViewController didSelectListItemAtIndexPath:]):
(-[WKSelectMenuListViewController cellForListItem:]): Deleted.
* Source/WebKit/UIProcess/ios/forms/WKTextInputListViewController.mm:
(-[WKTextInputListViewController viewDidLoad]):
* Source/WebKit/UIProcess/ios/forms/WKTimePickerViewController.mm:
(-[WKTimePickerViewController viewDidLoad]):

Canonical link: https://commits.webkit.org/275947@main
